std/math/powi.zig

```diff
@@ -0,0 +1,137 @@
+// Special Cases:
+//
+//  powi(x, +-0)    = 1 for any x
+//  powi(1, y)      = 1 for any y
+//  powi(x, 1)      = x for any x
+//  powi(nan, y)    = nan
+//  powi(x, nan)    = nan
+//  powi(+-0, y)    = +-inf for y an odd integer < 0
+//  powi(+-0, -inf) = +inf
+//  powi(+-0, +inf) = +0
+//  powi(+-0, y)    = +inf for finite y < 0 and not an odd integer
+//  powi(+-0, y)    = +-0 for y an odd integer > 0
+//  powi(+-0, y)    = +0 for finite y > 0 and not an odd integer
+//  powi(-1, +-inf) = 1
+//  powi(x, +inf)   = +inf for |x| > 1
+//  powi(x, -inf)   = +0 for |x| > 1
+//  powi(x, +inf)   = +0 for |x| < 1
+//  powi(x, -inf)   = +inf for |x| < 1
+//  powi(+inf, y)   = +inf for y > 0
+//  powi(+inf, y)   = +0 for y < 0
+//  powi(-inf, y)   = powi(-0, -y)
+//  powi(x, y)      = nan for finite x < 0 and finite non-integer y
+
+const builtin = @import("builtin");
+const std = @import("../index.zig");
+const math = std.math;
+const assert = std.debug.assert;
+const assertError = std.debug.assertError;
+
+inline fn checkFlow(comptime T: type, comptime info: builtin.TypeInfo, x: T, y: T) (error{Overflow, Underflow}!void) {
```

Comment:
It doesn't look like the semantics of this code require this function to be inline. If this function should be inline because it improves a benchmark then there should be a comment here explaining how to run the benchmark and the results with and without inlining. Otherwise functions should not be marked inline.